universal-application-tool-0.0.1/test/views/components/TextFormatterTest.java

```diff
@@ -12,17 +12,22 @@
   @Test
   public void urlsRenderCorrectly() {
     ImmutableList<DomContent> content =
-        TextFormatter.createLinksAndEscapeText("hello google.com http://internet.website");
+        TextFormatter.createLinksAndEscapeText(
+            "Hello google.com (http://internet.website), http://mysite.com...!");
 
-    assertThat(content).hasSize(4);
-    assertThat(content.get(0).render()).isEqualTo(new Text("hello ").render());
+    assertThat(content).hasSize(7);
+    assertThat(content.get(0).render()).isEqualTo(new Text("Hello ").render());
     assertThat(content.get(1).render())
         .isEqualTo("<a href=\"http://google.com/\" class=\"opacity-75\">google.com</a>");
-    assertThat(content.get(2).render()).isEqualTo(new Text(" ").render());
+    assertThat(content.get(2).render()).isEqualTo(new Text(" (").render());
     assertThat(content.get(3).render())
         .isEqualTo(
             "<a href=\"http://internet.website/\""
                 + " class=\"opacity-75\">http://internet.website</a>");
+    assertThat(content.get(4).render()).isEqualTo(new Text("), ").render());
```

Comment:
I am surprised this is accurately parsing the closed parenthesis ), since you don't include parenthesis in TextFormatter line 56. Do you know why this is passing? Perhaps I am missing something.

Reply:
Oh, I think this passes because UrlDetectorOptions.JAVASCRIPT catches the parenthesis and doesn't include them. Let me know if I'm wrong, or feel free to ignore if I'm right :D

Reply:
Yes - that's right. the JAVASCRIPT option looks for matched pairs of quotes, and various kinds of brackets and avoids including in the URL if it does find a matching pair.